elastic/eui
# [Emotion] Convert EuiDatePicker

## Summary

This PR converts `EuiDatePicker` and the underlying `react-datepicker` theme/custom styles to Emotion. For the latter, it heavily relies on nested 3rd party selectors - I consider modifying `react-datepicker` directly to use Emotion to be out of scope / unnecessary for now.

This is a fairly hefty PR and contains a ton of cleanup (there was a lot of unused CSS), so I strongly recommend [following along by commit](https://github.com/elastic/eui/pull/7937/commits) and relying primarily on QA.

## QA

The below components should generally look the same as production:
- [x] https://eui.elastic.co/pr_7937/storybook/?path=/story/forms-euidatepicker--playground
- [x] https://eui.elastic.co/pr_7937/storybook/?path=/story/forms-euidatepickerrange--playground
- Be sure to check out the new non-playground stories as well, as I set those up with several props that showcase certain modifiers/edge cases.

### General checklist

- Browser QA
- Docs site QA - N/A
- Code quality checklist
    ~- [ ] Added or updated **[jest](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/unit-testing.md) and [cypress](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/cypress-testing.md) tests**~
- Release checklist
    ~- [ ] If applicable, added the **breaking change** issue label (and filled out the breaking change checklist)~
- Designer checklist - N/A

<details><summary><b>Emotion checklist</b></summary>

**General**


---

**Unit tests**

- [x] Converted Enzyme to RTL

---

**Sass/Emotion conversion process**



---

**CSS tech debt**


---

**DOM Cleanup**

- [x] **SEARCH KIBANA FIRST**: Deleted any modifier classNames or maps if not being used in Kibana.  **No `euiDatePicker--` usages**

---

**Kibana due diligence**

  - ~[ ] If usage exists, consider converting to a `data` attribute so that consumers still have something to hook into~

---

**Extras/nice-to-have**

</details>
